src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/TensorsInfo.cs

```diff
@@ -15,16 +15,16 @@
 */
 
 using System;
+using System.Linq;
 using System.Collections.Generic;
-using Log = Tizen.Log;
 
 namespace Tizen.MachineLearning.Inference
 {
     /// <summary>
     /// The TensorsInfo class manages each Tensor information such as Name, Type and Dimension.
     /// </summary>
     /// <since_tizen> 6 </since_tizen>
-    public class TensorsInfo : IDisposable
+    public class TensorsInfo : IDisposable, IEquatable<TensorsInfo>
```

Comment:
Did you really need `TensorsInfor` class? can't it replace with a common collection class? like a `List<TensorInfo>` or `HashSet<TensorInfo>`, IEnumerable<TensorInfo>`?

even if need to a new class, implement collection interface like `IEnumerator<TensorInfo>`

``` c#

var tensorsInfo = new TensorsInfo();
tensorsInfo.AddTensorInfo("a", TensorType.Int8 ,  new int[] { 2 }); 
tensorsInfo.AddTensorInfo("b", TensorType.Int8 ,  new int[] { 2 }); 
tensorsInfo.AddTensorInfo("c", TensorType.Int8 ,  new int[] { 2 }); 

for (int i = 0; tensorsInfo.Count; i++) {
   Console.WriteLine($"Name : {tensorsInfo.GetTensorName(i)}, Dimension : {tensorsInfo.GetDimension(i)}, Size : {tensorsInfo.GetTensorSize(i)}");
 }

vs

var tensorsInfo = new TensorInfo[] {
  new TensorInfo("a", TensorType.Int8 ,  new int[] { 2 }),
  new TensorInfo("b", TensorType.Int8 ,  new int[] { 2 }),
  new TensorInfo("c", TensorType.Int8 ,  new int[] { 2 }),
}

foreach (TensorInfo tensor in tensorsInfo) {
   Console.WriteLine($"Name : {tensor.Name}, Dimension : {tensor.Dimension}, Size : {tensor.Size}");
}


```

Reply:
Becuase NNFramework (i.e. Backend ML engine) supports the multiple Tensors and its maximum count is **16**. `TensorsInfo` class is necessary to check its limitation. `AddTensorInfo()` in `TensorsInfo` class raise `IndexOutOfRangeException` when exceeding the size limits as below.

```c#
int idx = _infoList.Count;
if (idx >= Tensor.SizeLimit) {
    throw new IndexOutOfRangeException("Max size of the tensors is " + Tensor.SizeLimit);
}
_infoList.Add(new TensorInfo(name, type, dimension));
```
When using general collection instead of class, we cannot check the size limits.

Reply:
In case of 'TensorInfo' as public, Refer [this link](https://github.com/Samsung/TizenFX/pull/1154/files/cd49132137f5c954be0e53f4e731f12b8b49a592#r358032811).

Reply:
> 
> 
> Becuase NNFramework (i.e. Backend ML engine) supports the multiple Tensors and its maximum count is **16**. `TensorsInfo` class is necessary to check its limitation. `AddTensorInfo()` in `TensorsInfo` class raise `IndexOutOfRangeException` when exceeding the size limits as below.
> 
> ```cs
> int idx = _infoList.Count;
> if (idx >= Tensor.SizeLimit) {
>     throw new IndexOutOfRangeException("Max size of the tensors is " + Tensor.SizeLimit);
> }
> _infoList.Add(new TensorInfo(name, type, dimension));
> ```
> 
> When using general collection instead of class, we cannot check the size limits.


Hmm, We can check length of collection when that is used, for example in `public SingleShot(string modelAbsPath, TensorsInfo inTensorsInfo, TensorsInfo outTensorsInfo)` this API. 
Anyway, I also agree we do not want to change API